internal/tsc_wrapped/compiler_host.ts

```diff
@@ -359,6 +362,15 @@ export class CompilerHost implements ts.CompilerHost, tsickle.TsickleHost {
     return this.delegate.getDefaultLibFileName(options);
   }
 
+  realpath(s: string): string {
+    // tsc-wrapped relies on string matching of file paths for things like the
+    // file cache and for strict deps checking.
+    // TypeScript will try to resolve symlinks during module resolution which
+    // makes our checks fail.
+    // So we simply turn off symlink resolution.
```

Comment:
might want to add why we think this is safe. E.g. we have a flat node_modules tree, and so we don't usually expect to reach a single file in multiple ways?

This is also kind of annoying, the right fix on the TS side would be to use the inode...

Reply:
Looks like https://github.com/Microsoft/TypeScript/pull/12020 is closely related. They do symlink resolution only in this node_modules case, as you say because there are multiple ways to reach a module.
Don't we potentially have that as well? we don't always have a flat node_modules tree.